File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitDefects/EngineerVisitDefectsSQL.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace EngineerVisitDefectss

Public Class EngineerVisitDefectsSQL
    Private _database As Sys.Database

    Public Sub New(ByVal database As Sys.Database)
        _database = database
    End Sub

#Region "Functions"

	
    Public Sub Delete(ByVal EngineerVisitDefectID As Integer)
    		_database.ClearParameter()
            _database.AddParameter("@EngineerVisitDefectID", EngineerVisitDefectID, True)
            _database.ExecuteSP_NO_Return("EngineerVisitDefects_Delete")
    End Sub

    Public Function [EngineerVisitDefects_Get](ByVal EngineerVisitDefectID As Integer) As EngineerVisitDefects
        _database.ClearParameter()
        _database.AddParameter("@EngineerVisitDefectID", EngineerVisitDefectID)
        
        'Get the datatable from the database store in dt
        Dim dt As DataTable = _database.ExecuteSP_DataTable("EngineerVisitDefects_Get")
        
        If Not dt Is Nothing Then			
			if dt.rows.count > 0 then
				Dim oEngineerVisitDefects As New EngineerVisitDefects
				With oEngineerVisitDefects
				    .IgnoreExceptionsOnSetMethods = True
						.SetEngineerVisitDefectID= dt.Rows(0).Item("EngineerVisitDefectID")
				.SetCategoryID= dt.Rows(0).Item("CategoryID")
				.SetReason= dt.Rows(0).Item("Reason")
				.SetActionTaken= dt.Rows(0).Item("ActionTaken")
				.SetWarningNoticeIssued= dt.Rows(0).Item("WarningNoticeIssued")
				.SetDisconnected= dt.Rows(0).Item("Disconnected")
				.SetComments= dt.Rows(0).Item("Comments")
				.SetAssetID= dt.Rows(0).Item("AssetID")
				.SetEngineerVisitID= dt.Rows(0).Item("EngineerVisitID")

					.SetDeleted = dt.Rows(0).Item("Deleted")               
				End With		
				oEngineerVisitDefects.Exists = true				
			Return oEngineerVisitDefects
			else
					Return Nothing
			end if 
			       Else
            Return Nothing
        End If
            End Function

            Public Function EngineerVisitDefects_GetForEngineerVisit(ByVal EngineerVisitID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@EngineerVisitID", EngineerVisitID, True)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("EngineerVisitDefects_GetForEngineerVisit")
                dt.TableName = Entity.Sys.Enums.TableNames.tblEngineerVisitDefects.ToString
                Return New DataView(dt)
            End Function

            Public Function [EngineerVisitDefects_GetAll]() As DataView
                _database.ClearParameter()
                Dim dt As DataTable = _database.ExecuteSP_DataTable("EngineerVisitDefects_GetAll")
                dt.TableName = Entity.Sys.Enums.TableNames.tblEngineerVisitDefects.ToString
                Return New DataView(dt)
            End Function

            Public Function [Insert](ByVal oEngineerVisitDefects As EngineerVisitDefects) As EngineerVisitDefects
                _database.ClearParameter()
                AddEngineerVisitDefectsParametersToCommand(oEngineerVisitDefects)

                oEngineerVisitDefects.SetEngineerVisitDefectID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("EngineerVisitDefects_Insert"))
                oEngineerVisitDefects.Exists = True
                Return oEngineerVisitDefects
            End Function


            Public Sub [Update](ByVal oEngineerVisitDefects As EngineerVisitDefects)
                _database.ClearParameter()
                _database.AddParameter("@EngineerVisitDefectID", oEngineerVisitDefects.EngineerVisitDefectID, True)
                AddEngineerVisitDefectsParametersToCommand(oEngineerVisitDefects)
                _database.ExecuteSP_NO_Return("EngineerVisitDefects_Update")
            End Sub



            Private Sub AddEngineerVisitDefectsParametersToCommand(ByRef oEngineerVisitDefects As EngineerVisitDefects)
                With _database
                    .AddParameter("@CategoryID", oEngineerVisitDefects.CategoryID, True)
                    .AddParameter("@Reason", oEngineerVisitDefects.Reason, True)
                    .AddParameter("@ActionTaken", oEngineerVisitDefects.ActionTaken, True)
                    .AddParameter("@WarningNoticeIssued", oEngineerVisitDefects.WarningNoticeIssued, True)
                    .AddParameter("@Disconnected", oEngineerVisitDefects.Disconnected, True)
                    .AddParameter("@Comments", oEngineerVisitDefects.Comments, True)
                    .AddParameter("@AssetID", oEngineerVisitDefects.AssetID, True)
                    .AddParameter("@EngineerVisitID", oEngineerVisitDefects.EngineerVisitID, True)



                End With
            End Sub


#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitDefects/EngineerVisitDefectsSQL.cs
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVisitDefectss
    {
        public class EngineerVisitDefectsSQL
        {
            private Sys.Database _database;

            public EngineerVisitDefectsSQL(Sys.Database database)
            {
                _database = database;
            }

            public void Delete(int EngineerVisitDefectID)
            {
                _database.ClearParameter();
                _database.AddParameter("@EngineerVisitDefectID", EngineerVisitDefectID, true);
                _database.ExecuteSP_NO_Return("EngineerVisitDefects_Delete");
            }

            public EngineerVisitDefects EngineerVisitDefects_Get(int EngineerVisitDefectID)
            {
                _database.ClearParameter();
                _database.AddParameter("@EngineerVisitDefectID", EngineerVisitDefectID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("EngineerVisitDefects_Get");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oEngineerVisitDefects = new EngineerVisitDefects();
                        oEngineerVisitDefects.IgnoreExceptionsOnSetMethods = true;
                        oEngineerVisitDefects.SetEngineerVisitDefectID = dt.Rows[0]["EngineerVisitDefectID"];
                        oEngineerVisitDefects.SetCategoryID = dt.Rows[0]["CategoryID"];
                        oEngineerVisitDefects.SetReason = dt.Rows[0]["Reason"];
                        oEngineerVisitDefects.SetActionTaken = dt.Rows[0]["ActionTaken"];
                        oEngineerVisitDefects.SetWarningNoticeIssued = dt.Rows[0]["WarningNoticeIssued"];
                        oEngineerVisitDefects.SetDisconnected = dt.Rows[0]["Disconnected"];
                        oEngineerVisitDefects.SetComments = dt.Rows[0]["Comments"];
                        oEngineerVisitDefects.SetAssetID = dt.Rows[0]["AssetID"];
                        oEngineerVisitDefects.SetEngineerVisitID = dt.Rows[0]["EngineerVisitID"];
                        oEngineerVisitDefects.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oEngineerVisitDefects.Exists = true;
                        return oEngineerVisitDefects;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public DataView EngineerVisitDefects_GetForEngineerVisit(int EngineerVisitID)
            {
                _database.ClearParameter();
                _database.AddParameter("@EngineerVisitID", EngineerVisitID, true);
                var dt = _database.ExecuteSP_DataTable("EngineerVisitDefects_GetForEngineerVisit");
                dt.TableName = Sys.Enums.TableNames.tblEngineerVisitDefects.ToString();
                return new DataView(dt);
            }

            public EngineerVisitDefects Insert(EngineerVisitDefects oEngineerVisitDefects)
            {
                _database.ClearParameter();
                AddEngineerVisitDefectsParametersToCommand(ref oEngineerVisitDefects);
                oEngineerVisitDefects.SetEngineerVisitDefectID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("EngineerVisitDefects_Insert"));
                oEngineerVisitDefects.Exists = true;
                return oEngineerVisitDefects;
            }

            public void Update(EngineerVisitDefects oEngineerVisitDefects)
            {
                _database.ClearParameter();
                _database.AddParameter("@EngineerVisitDefectID", oEngineerVisitDefects.EngineerVisitDefectID, true);
                AddEngineerVisitDefectsParametersToCommand(ref oEngineerVisitDefects);
                _database.ExecuteSP_NO_Return("EngineerVisitDefects_Update");
            }

            private void AddEngineerVisitDefectsParametersToCommand(ref EngineerVisitDefects oEngineerVisitDefects)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@CategoryID", oEngineerVisitDefects.CategoryID, true);
                    withBlock.AddParameter("@Reason", oEngineerVisitDefects.Reason, true);
                    withBlock.AddParameter("@ActionTaken", oEngineerVisitDefects.ActionTaken, true);
                    withBlock.AddParameter("@WarningNoticeIssued", oEngineerVisitDefects.WarningNoticeIssued, true);
                    withBlock.AddParameter("@Disconnected", oEngineerVisitDefects.Disconnected, true);
                    withBlock.AddParameter("@Comments", oEngineerVisitDefects.Comments, true);
                    withBlock.AddParameter("@AssetID", oEngineerVisitDefects.AssetID, true);
                    withBlock.AddParameter("@EngineerVisitID", oEngineerVisitDefects.EngineerVisitID, true);
                }
            }
        }
    }
}